--- a/nginx.conf.sigil
+++ b/nginx.conf.sigil
@@ -57,6 +57,8 @@
     gzip_comp_level  6;
 
     proxy_pass  http://{{ $.APP }}-{{ $upstream_port }};
+    proxy_intercept_errors on;
+    error_page 404 = @public_html;
     {{ if and $is_ssl (eq $.HTTP2_PUSH_SUPPORTED "true") }}http2_push_preload on; {{ end }}
     proxy_http_version 1.1;
     proxy_connect_timeout {{ $.PROXY_CONNECT_TIMEOUT }};
@@ -110,6 +112,11 @@
     internal;
   }
   {{ end }}
+
+  location @public_html {
+    root {{ var "PUBLIC_HTML_DIR" }};
+  }
+
   include {{ $.DOKKU_ROOT }}/{{ $.APP }}/nginx.conf.d/*.conf;
 {{ end }}
 }
